apps/glusterfs/cluster_entry_test.go

```diff
@@ -380,3 +380,79 @@ func TestNewClusterEntryNewInfoResponse(t *testing.T) {
 	tests.Assert(t, reflect.DeepEqual(info.Nodes, c.Info.Nodes))
 	tests.Assert(t, reflect.DeepEqual(info.Volumes, c.Info.Volumes))
 }
+
+func TestUpdateClusterEntryForFlags(t *testing.T) {
+	tmpfile := tests.Tempfile()
+	defer os.Remove(tmpfile)
+
+	// Create the app
+	app := NewTestApp(tmpfile)
+	//logger.SetLevel(utils.LEVEL_DEBUG)
+
+	// Create a cluster
+	cluster_req := &api.ClusterCreateRequest{
+		File: true,
+	}
+	c := NewClusterEntryFromRequest(cluster_req)
+	c.NodeAdd("node_abc")
+	c.NodeAdd("node_def")
+
+	// Save element in database
+	err := app.db.Update(func(tx *bolt.Tx) error {
+		return c.Save(tx)
+	})
+	tests.Assert(t, err == nil)
+
+	//Read the cluster info again and verify flags
+	var info *api.ClusterInfoResponse
+	err = app.db.View(func(tx *bolt.Tx) error {
+		cluster, err := NewClusterEntryFromId(tx, c.Info.Id)
+		if err != nil {
+			return err
+		}
+
+		info, err = cluster.NewClusterInfoResponse(tx)
+		if err != nil {
+			return err
+		}
+		logger.Info("file flag: %v, block flag %v", info.File, info.Block)
+		return nil
+	})
+	tests.Assert(t, err == nil)
+	tests.Assert(t, info.File == true)
+	tests.Assert(t, info.Block == false)
+
+	// remove the update flag from db
+	err = app.db.Update(func(tx *bolt.Tx) error {
+		dbaentry, err := NewDbAttributeEntryFromKey(tx, DB_HAS_FILE_BLOCK_FLAG)
+		if err != nil {
+			return err
+		}
+		return dbaentry.Delete(tx)
+	})
+	tests.Assert(t, err == nil)
+	app.Close()
+
+	// Recreate the app
```

Comment:
The comment should mention that this triggers the upgrade path (that's the point).

Reply:
done